quickwit-cli/src/lib.rs

```diff
@@ -220,15 +223,40 @@ pub async fn index_data_cli(args: IndexDataArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
-fn create_source_config_from_args(input_path_opt: Option<PathBuf>) -> SourceConfig {
-    let params = FileSourceParams {
-        filepath: input_path_opt,
-    };
-    SourceConfig {
-        id: "cli_source".to_string(),
-        source_type: "file".to_string(),
-        params: serde_json::to_value(params).unwrap(),
+async fn create_source_config_from_args(
+    source_config_path_opt: Option<&PathBuf>,
+    input_path_opt: Option<&PathBuf>,
+) -> anyhow::Result<SourceConfig> {
+    if let Some(source_config_path) = source_config_path_opt {
+        let source_config_file = File::open(source_config_path).with_context(|| {
```

Comment:
nitpick.
`fs::read` works fine for small files. It avoid building the `BufReader`.

Another nice thing is that your entirely separate the io from the deserialization, making the source of the error clearer.
https://doc.rust-lang.org/stable/std/fs/fn.read.html